packages/core/src/domain/configuration/transportConfiguration.ts

```diff
@@ -1,48 +1,56 @@
 import { BuildEnv, BuildMode } from '../../boot/init'
+import { objectValues } from '../../tools/utils'
 import { InitConfiguration } from './configuration'
-import { createEndpointBuilder, ENDPOINTS_TYPES, INTAKE_SITE_US } from './endpointBuilder'
+import { createEndpointBuilder, INTAKE_SITE_US, EndpointBuilder } from './endpointBuilder'
 
 export interface TransportConfiguration {
-  logsEndpoint: string
-  rumEndpoint: string
-  sessionReplayEndpoint: string
-  internalMonitoringEndpoint?: string
-  isIntakeUrl: (url: string) => boolean
-
+  logsEndpointBuilder: EndpointBuilder
+  rumEndpointBuilder: EndpointBuilder
+  sessionReplayEndpointBuilder: EndpointBuilder
+  internalMonitoringEndpointBuilder?: EndpointBuilder
+  isIntakeEndpoint: (url: string) => boolean
   // only on staging build mode
   replica?: ReplicaConfiguration
 }
 
 export interface ReplicaConfiguration {
   applicationId?: string
-  logsEndpoint: string
-  rumEndpoint: string
-  internalMonitoringEndpoint: string
+  logsEndpointBuilder: EndpointBuilder
+  rumEndpointBuilder: EndpointBuilder
+  internalMonitoringEndpointBuilder: EndpointBuilder
 }
 
 export function computeTransportConfiguration(
   initConfiguration: InitConfiguration,
   buildEnv: BuildEnv
 ): TransportConfiguration {
-  const endpointBuilder = createEndpointBuilder(initConfiguration, buildEnv)
-  const intakeUrls: string[] = ENDPOINTS_TYPES.map((endpointType) => endpointBuilder.buildIntakeUrl(endpointType))
+  const endpointBuilders = {
+    logsEndpointBuilder: createEndpointBuilder(initConfiguration, buildEnv, 'logs'),
+    rumEndpointBuilder: createEndpointBuilder(initConfiguration, buildEnv, 'rum'),
+    sessionReplayEndpointBuilder: createEndpointBuilder(initConfiguration, buildEnv, 'sessionReplay'),
+  }
+  const intakeEndpoints: string[] = objectValues(endpointBuilders).map((builder) => builder.buildIntakeUrl())
 
   const configuration: TransportConfiguration = {
-    isIntakeUrl: (url: string) => intakeUrls.some((intakeUrl) => url.indexOf(intakeUrl) === 0),
-    logsEndpoint: endpointBuilder.build('logs'),
-    rumEndpoint: endpointBuilder.build('rum'),
-    sessionReplayEndpoint: endpointBuilder.build('sessionReplay'),
+    isIntakeEndpoint: (url) => intakeEndpoints.some((intakeEndpoint) => url.indexOf(intakeEndpoint) === 0),
```

Comment:
I found `isIntakeUrl` more accurate.
With this naming, I would expect it to the check that `url === intakeEndpoint`